basenode/src/main/java/io/coti/basenode/services/BaseNodeTransactionSynchronizationService.java

```diff
@@ -75,7 +74,8 @@ private ResponseExtractor<Void> getResponseExtractorForMissingTransactionChunks(
                         receivedMissingTransactionNumber.incrementAndGet();
                         if (!insertMissingTransactionThread.isAlive()) {
                             missingTransactionExecutorMap = new EnumMap<>(InitializationTransactionHandlerType.class);
-                            EnumSet.allOf(InitializationTransactionHandlerType.class).forEach(initializationTransactionHandlerType -> missingTransactionExecutorMap.put(initializationTransactionHandlerType, new ExecutorData(initializationTransactionHandlerType)));
+                            EnumSet.allOf(InitializationTransactionHandlerType.class).forEach(initializationTransactionHandlerType ->
```

Comment:
this will work assuming it is not processing and reading new missed TX in parallel

Reply:
assumption made was wrong. code is doing parallel work on missing tx , but since there is a blocking loop in missing waiting for all TX from recovery, this part works.